### PR TITLE
[DX-1813] Upgrade node matrix for Node 18

### PR DESCRIPTION
**Pull request by matthewmuscat**
https://immutable.atlassian.net/browse/DX-1813

# Summary
Allowing for Node 18 when consuming and also for testing workflows. 

# Why the changes
Customers want it! https://immutable.atlassian.net/browse/DX-1813

# Things worth calling out
Since this repo has a [dependency of itself](https://github.com/immutable/imx-core-sdk/blob/main/package.json#L92), [the test suite using Node 18 ](https://github.com/immutable/imx-core-sdk/actions/runs/4559401698/jobs/8043316873?pr=293)won't pass as the existing package doesn't allow for Node 18. 

Therefore I'm going to remove this CI test, and once we re-publish the package, we can add this back. To verify this upgrade won't break anything, we can locally run the tests and build/test with yalc.

# Before merging
- [ ] ***For Immutable developers:*** Do the changes in this PR require documentation updates? Please refer to this [SDK documentation guide](https://immutable.atlassian.net/wiki/spaces/PPS/pages/1916994017/SDK+documentation+guide) and list links to documentation update PRs (they don't have to be merged) below (or write `N/A`):
    - [ ] *Add documentation update 1*
    - [ ] *Add documentation update 2*

### CODE REVIEW

**Comment by matthewmuscat** [comment]
Since this repo has a [dependency of itself](https://github.com/immutable/imx-core-sdk/blob/main/package.json#L92), the test suite using Node 18 won't pass as the existing package doesn't allow for Node 18. 

Therefore I'm going to remove this CI test, and once we re-publish the package, we can add this back. To verify this upgrade won't break anything, we can locally run the tests and build/test with yalc.

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
   "license": "Apache-2.0",
   "bugs": "https://github.com/immutable/imx-core-sdk/issues",
   "engines": {
-    "node": ">=14.0.0 <17.0.0"
+    "node": ">=14.0.0 <19.0.0"
```
Comment (cajames) [inline package.json:26]: Is there a reason to have the upperbound limit?

Comment (matthewmuscat) [reply package.json:26]: Yeah it can be, wasn't sure if we strictly wanted nothing above v18.x as v19 is the "LTS"?

Comment (matthewmuscat) [reply package.json:26]: Sure thing, I do agree, I wasn't sure if this was already considered in the ticket as the A/C was to allow for Node 18 specifically. Happy to remove the upper bound and we can revisit if there's any issues with later versions

Comment (cajames) [reply package.json:26]: Yep please do

**Review by suluvai — APPROVED** [review]

**Review by cajames — APPROVED** [review]

**Pull request merged**